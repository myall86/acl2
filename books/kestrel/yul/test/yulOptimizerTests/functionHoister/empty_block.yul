{
    let a
    { }
    function f() -> x {
        let b := 4
        { }
        for {} f() {} {}
    }
}
// ====
// dialect: yul
// ----
// step: functionHoister
//
// {
//     let a
//     function f() -> x
//     {
//         let b := 4
//         for { } f() { }
//         { }
//     }
// }
