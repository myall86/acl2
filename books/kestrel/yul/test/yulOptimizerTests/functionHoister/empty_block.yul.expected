
 {
     let a
     function f() -> x
     {
         let b := 4
         for { } f() { }
         { }
     }
 }
